src/lib.rs

```diff
@@ -48,5 +53,55 @@
 #![cfg_attr(feature="clippy", deny(clippy, unicode_not_nfc, wrong_pub_self_convention,
                                    option_unwrap_used))]
 
-#[test]
-fn place_holder() {}
+// `error_chain!` can recurse deeply
+#![recursion_limit = "1024"]
+
+#[macro_use]
+extern crate error_chain;
+
+
+#[cfg(target_os = "macos")]
+extern crate libc;
+
+mod app;
+pub use app::App;
+
+mod errors {
+    error_chain! {
+        types {
+            Error, ErrorKind, ChainErr, Result;
+        }
+
+        errors {
+            /// The SystemURI error used to wrap problems
+            SystemUriError(t: String) {
+                description("System URI Erro")
+                display("Could not execute: {}", t)
+            }
+        }
+    }
+}
+
+pub use errors::Error as SystemUriError;
+
+
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "linux")]
+pub use linux::{install, open};
+
+
+#[cfg(target_os = "windows")]
+mod windows;
+#[cfg(target_os = "windows")]
+pub use windows::{install, open};
+
+
+#[cfg(target_os = "macos")]
+mod macos;
+#[cfg(target_os = "macos")]
+pub use macos::{install, open};
+
+/// Foreign Function call Interface to use this library
+#[cfg(feature="ffi")]
```

Comment:
the feature is _on by default_. but FFI requires libc now, which as a standalone-lib is not needed for neither windows nor linux. So if I wanted to use it from a rust project and I don't want extra libc stuff, I could deactivate it in my deps config. That is all the thinking here, no harm done.